lib/language_pack/ruby.rb

```diff
@@ -694,19 +694,32 @@ def build_bundler
       end
 
       if bundler.windows_gemfile_lock?
-        log("bundle", "has_windows_gemfile_lock")
-
-        File.unlink("Gemfile.lock")
-        ENV.delete("BUNDLE_DEPLOYMENT")
-
-        warn(<<~WARNING, inline: true)
-          Removing `Gemfile.lock` because it was generated on Windows.
-          Bundler will do a full resolve so native gems are handled properly.
-          This may result in unexpected gem versions being used in your app.
-          In rare occasions Bundler may not be able to resolve your dependencies at all.
-
-          https://devcenter.heroku.com/articles/bundler-windows-gemfile
-        WARNING
+        if bundler.supports_multiple_platforms?
+          puts "Windows `Gemfile.lock` detected, bundler #{@bundler.version} supports multiple platforms, no action taken."
```

Comment:
"No action taken": This text can be confusing, to people who don't know the history.
I think it is better to simply not print anything.

Reply:
I have an output goal that is "anytime there's a change in behavior based on buildpack logic, developers should be able to tell by comparing the outputs." 

Humans are also bad at seeing "what's not there." I.e. If one build shows "Installing mail" but it's not in the other build, most people usually have to use diff tools to determine notice unless they're explicitly looking for that line.

Another goal is that if they have a failure related to that different action they should be able to trace that action in the same build. If someone's app started to fail they can scan up and might see that line and be curious only to discover that behavior is the cause of their problems and there's a bug in bundler.

So it's a little aggressive to log in the empty if case but I feel like it's a possibly VERY large change to the windows users and they deserve to know about it.

Now about "no action taken." If that phrase is confusing then we can remove it later or ask users what might be better. What I'm trying to avoid with that line is when you show people something, they want to know what happens next because of that thing.

You can see this with a storytelling narrative principle called "checkov's gun" which loosely is "if a gun is introduced in the first act it should go off in the third." Here's a funny video that is satire but demonstrates the principle https://www.youtube.com/watch?v=PQOvv_b9TpY. The viewer keeps waiting for it to "go off."

The problem with saying simply:

```
"Windows `Gemfile.lock` detected, bundler #{@bundler.version} supports multiple platforms"
```

Is that the reader will implicitly be waiting for the seemingly missing information. They will be thinking "why did you tell me that" as well as "well, what will you do with that information?"

The current output might be a little curious or weird. Maybe even eyebrow-raising. But it won't be anxiety or stress-inducing.

So I agree it's off, but it's also purposeful in it's off-ness.

Reply:
It is a good point. I think something like this will ensure that people who was not aware of the previous behavior and notice the message are not confused and use time on investigating.
```suggestion
          puts "Windows `Gemfile.lock` detected, bundler #{@bundler.version} supports multiple platforms, no action taken (before June 25th, Heroku would have removed the lock file)."
```